packages/launch/src/focused-launch/domain-details/index.tsx

```diff
@@ -3,24 +3,92 @@
 import { __ } from '@wordpress/i18n';
+import DomainPicker, { ITEM_TYPE_BUTTON } from '@automattic/domain-picker';
+import type { DomainSuggestions } from '@automattic/data-stores';
+import { Title, SubTitle } from '@automattic/onboarding';
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { Icon, chevronLeft } from '@wordpress/icons';
 
 /**
  * Internal dependencies
  */
 import { Route } from '../route';
+import { useSite, useDomainSearch } from '../../hooks';
+import { LAUNCH_STORE } from '../../stores';
+import { FOCUSED_LAUNCH_FLOW_ID } from '../../constants';
 
 import './style.scss';
 
-const DomainDetails: React.FunctionComponent = () => {
+const ANALYTICS_UI_LOCATION = 'domain_step';
+
+const DomainStep: React.FunctionComponent = () => {
+	const { plan, domain } = useSelect( ( select ) => select( LAUNCH_STORE ).getState() );
+	const { currentDomainName } = useSite();
+	const domainSearch = useDomainSearch();
+
+	const {
+		setDomain,
+		unsetDomain,
+		setDomainSearch,
+		unsetPlan,
+		confirmDomainSelection,
+	} = useDispatch( LAUNCH_STORE );
+
+	const handleDomainSelect = ( suggestion: DomainSuggestions.DomainSuggestion ) => {
+		confirmDomainSelection();
+		setDomain( suggestion );
+		if ( plan?.isFree ) {
+			unsetPlan();
+		}
+	};
+
+	const handleExistingSubdomainSelect = () => {
+		unsetDomain();
+	};
+
+	const trackDomainSearchInteraction = ( query: string ) => {
+		recordTracksEvent( 'calypso_newsite_domain_search_blur', {
+			flow: FOCUSED_LAUNCH_FLOW_ID,
+			query,
+			where: ANALYTICS_UI_LOCATION,
+		} );
+	};
+
 	return (
 		<div>
-			<Link to={ Route.Summary }>{ __( 'Go back', __i18n_text_domain__ ) }</Link>
-			<p>{ __( 'Choose a domain', __i18n_text_domain__ ) }</p>
+			<div className="focused-launch-domain-details__header">
+				<div className="focused-launch-domain-details__back-link">
+					<Link to={ Route.Summary }>
+						<Icon icon={ chevronLeft } /> <span>{ __( 'Go back', __i18n_text_domain__ ) }</span>
+					</Link>
+				</div>
+				<Title>{ __( 'Choose a domain', __i18n_text_domain__ ) }</Title>
+				<SubTitle>
+					{ __( 'Free for the first year with any paid plan.', __i18n_text_domain__ ) }
+				</SubTitle>
+			</div>
+			<div className="focused-launch-domain-details__body">
+				<DomainPicker
+					initialDomainSearch={ domainSearch }
+					onSetDomainSearch={ setDomainSearch }
+					onDomainSearchBlur={ trackDomainSearchInteraction }
+					currentDomain={ domain?.domain_name || currentDomainName }
+					existingSubdomain={ currentDomainName }
+					onDomainSelect={ handleDomainSelect }
+					onExistingSubdomainSelect={ handleExistingSubdomainSelect }
+					analyticsFlowId={ FOCUSED_LAUNCH_FLOW_ID }
+					analyticsUiAlgo={ ANALYTICS_UI_LOCATION }
+					segregateFreeAndPaid
+					locale={ document.documentElement.lang }
+					itemType={ ITEM_TYPE_BUTTON }
+				/>
+			</div>
 		</div>
 	);
 };
 
-export default DomainDetails;
+export default DomainStep;
```

Comment:
More of a nitpick, but can we keep calling this view "domain details" or find a name that is different from "step"? Just so that we can all agree on a naming convention.

So far, what I went with is the following:
- The focused launch modal has a router. Each route is associated to a _view_, e.g. Summary, DomainDetails, PlanDetails
- Within the Summary View, there are _steps_: the site title step, the domain step, and the plan step

If this is confusing, we can discuss and change the code / PR descriptions accordingly :)

Reply:
This is a good point. I'll definitely make that change.